docs/lib/introduction.md

```diff
@@ -0,0 +1,789 @@
+2) The `BlobHandlerRegistrationHandlerRegistry` selects the `BlobHanderRegistrationHandler` at `"Plugin/Test"` and
+   registers the corresponding ("anonymous") BlobHandler at its corresponding Registry.
+
+The current standard implementation of the `HandlerRegistrationHandlerRegistry` uses variant 2), therefore, it always
+resolves the _handler name_ with the **longest matching path** within its hierarchical namespace.
+
+As a `HandlerRegistrationHandlerRegistry` is a `HandlerRegistrationHandler` (see
+[interface](../../pkg/registrations/registrations.go)), a `HandlerRegistrationHandlerRegistry` could theoretically also
+contain other`HandlerRegistrationHandlerRegistries`.
+
+This _register by name_ mechanism is especially also used to register _plugin implementations_ of _handlers_ at their
+corresponding _registries_ ([here](../../pkg/contexts/ocm/blobhandler/generic/plugin/registration.go) is the
+`HandlerRegistrationHandler` for _plugin implementations_ of `BlobHandler`).
+
+### Section 3.2.3 - Repository Spec Handlers
+
+The `RepositorySpecHandlers` can be considered a _Constraint Registry_ providing _Handlers_ for converting the so
+called _UniformRepositorySpecs_ (a uniform string representation for RepositorySpecs) into the repository type
+specific _RepositorySpec_ representation. Therefore, its constraint merely consists of the _Repository Type_.  
+As it is rather cumbersome to enter JSON objects such as a RepositorySpec through the command line and as there may be
+multiple types of Repository Specs for certain Repository Types (e.g. legacy types), the ocm-cli allows to enter a
+uniform string representation called `UniformRepositorySpec` of a Repository Spec. Be aware, there are again different
+UniformRepositorySpecs for [OCM Repositories](../../pkg/contexts/ocm/internal/uniform.go) and for
+[OCI Repositories](../../pkg/contexts/oci/internal/uniform.go). Furthermore, _ocm_ and _oci references_, or rather
+_reference strings_ (e.g. `eu.gcr.io/gardener-project/landscaper/examples/charts/hello-world:1.0.0`) are parsed into a
+[representation that is based on a `UniformRepositorySpec`](../../pkg/contexts/oci/ref.go). Thus, to get access to the
+respective Repository through the ocm-lib, this `UniformRepositorySpec` first has to be converted into a corresponding
+`RepositorySpec`.  
+In conclusion, a `RepositorySpecHandlers` _Constraint Registry_ is an object that maps a _repository type
+(= constraint)_ to a number of corresponding `RepositorySpecHandler` objects that might be able to convert a
+`UniformRepositorySpec` into the appropriate `RepositorySpec` for that _Repository Type_. Therefore, a
+`RepositorySpecHandler` has to implement the following [interface](../../pkg/contexts/ocm/internal/uniform.go):
+
+```
+type RepositorySpecHandler interface {
+   MapReference(ctx Context, u *UniformRepositorySpec) (RepositorySpec, error)
+}
+```
+
+A `RepositorySpecHandlers` Registry **is not** also a `HandlerRegistrationHandlerRegistry`.
```

Comment:
Why is it necessary to state this so explicitly here ? And if it is necessary, some reasoning why this is not the case should be provided.